node_modules/gh-users/lib/internal/dao.js

```diff
@@ -370,20 +370,26 @@ var getUserCalendar = module.exports.getUserCalendar = function(user, includeCon
         dateTimeFilter.end = {'lte': end};
     }
 
+    // This query fetches such a massive amount of data, that the CPU usage
```

Comment:
It's hard to compare tests as each run would add more series and modules to a user's calendar which would make subsequent runs slower. That said, between two runs I saw a drop of about 250ms on a mean of ~500ms. So roughly half.

I'll try to have a model loaded where each user has a set of subscribed series/modules and write a read-only performance test.